### PR TITLE
Add SBRPs for Roslyn

### PR DESCRIPTION
**Pull request by mmitche**

### CODE REVIEW

**Comment by mthalman** [comment]
For System.Collections.Immutable, you're running into https://github.com/dotnet/sdk/issues/31570. I suspect there will be many more errors showing up. See https://github.com/dotnet/sdk/issues/31570#issuecomment-1498804761 for workaround.

File: eng/Build.props [hunk 1]
```diff
@@ -83,10 +83,22 @@
     <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\Microsoft.Build.Utilities.Core.15.1.1012.csproj" />
 
     <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\Microsoft.Bcl.AsyncInterfaces.7.0.0.csproj" />
+    <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\System.IO.Pipelines.7.0.0.csproj" />
```
Comment (MichaelSimons) [inline eng/Build.props:86]: Any new packages should be grouped at the end of this file per this documentation - https://github.com/dotnet/source-build-reference-packages/pull/612/files#diff-366b25472963c6b33188d84139f6c8997298c5adc857a2624ab0d26a2d87baf0L15.  Not a big deal for this particular change but it is the recommended practice with a valid reason.

**Review by MichaelSimons — APPROVED** [review]

File: eng/Build.props (hunk 1 above)
Comment (mmitche) [reply eng/Build.props]: Okay thanks. I tried to add them where I thought was logical. Will extract to the end.

Comment (MichaelSimons) [reply eng/Build.props]: I already fixed it for you.

**Pull request merged**